src/lib/addons/slack-app.ts

```diff
@@ -17,64 +21,103 @@ interface ISlackAppAddonParameters {
+            }
+
+            const taggedChannels = this.findTaggedChannels(event);
+
+            if (this.slackChannels?.length && taggedChannels.length) {
+                const slackChannelsToPostTo = this.slackChannels.filter(
+                    ({ id, name }) =>
+                        id && name && taggedChannels.includes(name),
+                );
 
-        const slackChannels = await this.slackClient.conversations.list({
-            types: 'public_channel,private_channel',
-        });
-        const taggedChannels = this.findTaggedChannels(event);
-
-        if (slackChannels.channels?.length && taggedChannels.length) {
-            const slackChannelsToPostTo = slackChannels.channels.filter(
-                ({ id, name }) => id && name && taggedChannels.includes(name),
-            );
-
-            const text = this.msgFormatter.format(event);
-            const featureLink = this.msgFormatter.featureLink(event);
-
-            const requests = slackChannelsToPostTo.map(({ id }) =>
-                this.slackClient!.chat.postMessage({
-                    channel: id!,
-                    text,
-                    attachments: [
-                        {
-                            actions: [
-                                {
-                                    name: 'featureToggle',
-                                    text: 'Open in Unleash',
-                                    type: 'button',
-                                    value: 'featureToggle',
-                                    style: 'primary',
-                                    url: featureLink,
-                                },
-                            ],
-                        },
-                    ],
-                }),
-            );
-
-            await Promise.all(requests);
-            this.logger.info(`Handled event ${event.type}.`);
+                const text = this.msgFormatter.format(event);
+                const featureLink = this.msgFormatter.featureLink(event);
+
+                const requests = slackChannelsToPostTo.map(({ id }) =>
+                    this.slackClient!.chat.postMessage({
+                        channel: id!,
+                        text,
+                        attachments: [
+                            {
+                                actions: [
+                                    {
+                                        name: 'featureToggle',
+                                        text: 'Open in Unleash',
+                                        type: 'button',
+                                        value: 'featureToggle',
+                                        style: 'primary',
+                                        url: featureLink,
+                                    },
+                                ],
+                            },
+                        ],
+                    }),
+                );
+
+                await Promise.all(requests);
```

Comment:
I'm wondering whether we should use `Promise.allSettled` here and just map rejected promises to a warning message, rather than handling an exception if any of the promises failed.

Reply:
Nice! Did a major refactor that addresses this in https://github.com/Unleash/unleash/pull/4284/commits/4883ea1eade9e938c1d8246a681273ad35b1af30

Reply:
Looks nice!